base/iterator.jl

```diff
@@ -210,3 +210,11 @@ next(it::Repeated, state) = (it.x, nothing)
 done(it::Repeated, state) = false
 
 repeated(x, n::Int) = take(repeated(x), n)
+
+# Zips with infinite length components
+
+length{I<:Union{Cycle,Repeated},Z<:AbstractZipIterator}(z::Zip{I,Z}) = length(z.z)
```

Comment:
How's about at the top we do

    typealias InfiniteIterators Union{Cycle,Repeated}

and add a TODO that we should do this with traits once we have #13222.